### PR TITLE
docs: adjust base image to ubuntu20.04

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Likely we have been using "strange" Python version as this does not appear with other CI checks
see: https://bobbyhadz.com/blog/python-importerror-cannot-import-name-mapping-from-collections

Fixes https://github.com/Lightning-AI/lightning/pull/17751#issuecomment-1594025293

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @tchaton

### CODE REVIEW

**Comment by Borda** [comment]
reporting in https://github.com/aws-actions/configure-aws-credentials/issues/747

**Comment by Borda** [comment]
Conclusion: our custom base docs image was switched to Ubuntu 20.04 to match the readthedocs 
https://github.com/Lightning-AI/lightning/blob/c7a5f0604691ee0126cf1fb53498e2062b0b1599/.readthedocs.yml#L32
Also, ubuntu 22.04 has default py3.10, and for some docs requirements intersection we have installed quite only AWS-CLI package, which has not addressed this issue yet, so it is possible with bumping some dependencies, we may resolve this problem as bonus side effect

File: .github/workflows/ci-dockers.yml
Comment (Borda) [inline .github/workflows/ci-dockers.yml:1]: this was renamed as it includes all images including building docs base image

**Review by justusschock — APPROVED** [review]

File: .github/workflows/ci-dockers.yml
Comment (Borda) [reply .github/workflows/ci-dockers.yml:1]: updated

**Review by carmocca — APPROVED** [review]

**Pull request merged**